sycl/include/CL/sycl/detail/kernel_program_cache.hpp

```diff
@@ -78,6 +82,7 @@ class KernelProgramCache {
 
   ProgramCacheT MCachedPrograms;
   KernelCacheT MKernelsPerProgramCache;
+  PlatformImplPtr MPlatform;
```

Comment:
It seems very strange to have platform object in the cache.
@s-kanaev Do you know if we can do better?

Reply:
> It seems very strange to have platform object in the cache.
> @s-kanaev Do you know if we can do better?

plugin_impl is only kept in Platform object.
To access it, platform_impl has been included here. The platform_impl here is from the context_impl to which the cache belongs.
Will it be better to include a reference to the parent Context? I wasn't able to achieve it in the code. So i decided to pass the platform the context belongs to.

Reply:
> As far as I understand cache only requires plugin information, not the whole platform. I'd prefer to keep plugin reference/pointer instead of platform.

Yes, only the plugin information is needed. The overall design takes care that only platform_impl owns a shared_ptr to the plugin, and others get a reference to the plugin only through platform_impl class. Storing a reference to plugin_impl has not been done anywhere. So I refrained from storing it here.

> Besides., is it even possible to retrieve plugin information out of PI kernel/program object/pointer?

No we cannot access the plugin_information through any PI class objects. Atleast not with the current design.

Reply:
> Will it be better to include a reference to the parent Context? I wasn't able to achieve it in the code. So i decided to pass the platform the context belongs to.


This may be quite reasonable.
One may store a reference to parent context as well as a mere pointer. Besides, pointer will be wrapped with weak_ptr.

Reply:
hmmm.. Let me try.. I'll update the code.